src/metrics/representation.py

```diff
@@ -0,0 +1,381 @@
+        return generator
+
+    def prepare_representation_data(self, vectors: np.ndarray, queries_idxs: np.ndarray, scores: np.ndarray
+    ) -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+        """Prepare data for faiss build index, and following search, in representation dataset case.
+        
+        Separate query and database vectors from storage vectors.
+        Prepare scores.
+        Generate relevant indexes for every query request.
+
+        Args:
+            vectors: Vectors of all storage, include queries and database vectors, size (database_size, embedding_size).
+            queries_idxs: Bool array wich indicate if vector is query, size (database_size).
+            scores: Array of scores for every relevant index per each query request, size (database_size, query_size).
+                See representation dataset for more information.
+
+        Retruns:
+            q_vecs: Queries vectors, size (queries_size, embedding_size).
+            db_vecs: Database vectors, for faiss build index, size (database_size, embedding_size).
+            relevant: Array of relevant indexes in database for every query request, size (queries_size, ).
+            scores: Array of scores without queries empty scores.
+            db_idxs: Array with all database indexes.
+            queries_idxs: Array of queries order number.
+        """
+        is_queries = queries_idxs >= 0
+        queries_idxs = queries_idxs[is_queries]
+        q_vecs = vectors[is_queries]
+        db_vecs = vectors[~is_queries]
+        db_idxs = np.arange(len(db_vecs))
+        scores = scores[~is_queries]
+        
+        relevant = []
+        empty_relevant_idxs = []
+        for idx in range(len(q_vecs)):
+            relevant_idxs = np.where(scores[:, queries_idxs[idx]] > 0.)[0]
+            if len(relevant_idxs) == 0:
+                empty_relevant_idxs.append(idx)
+            else:
+                relevant.append(relevant_idxs)
+        relevant = np.array(relevant)
+        
+        # remove empty relevant queries
+        q_vecs = np.delete(q_vecs, empty_relevant_idxs, axis=0)
+        queries_idxs = np.delete(queries_idxs, empty_relevant_idxs)
+
+        return q_vecs, db_vecs, relevant, scores, db_idxs, queries_idxs
+
+    def prepare_classification_data(self, vectors: np.ndarray, targets: np.ndarray
+    ) -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+        """Prepare data for faiss build index, and following search, in classification dataset case.
+        
+        Separate query and database vectors from storage vectors.
+        Query vectors index would be first found target uniq value in targets array. Query size is uniq target count.
+
+        Args:
+            vectors: Vectors of all storage, include queries and database vectors, size (database_size, embedding_size).
+            targets: Targets in classification task for every vectors, size (database_size).
```

Comment:
```suggestion
            targets: Targets in classification task for every vector, size (database_size)
```
Also, add range for values